setup.ts

```diff
@@ -279,48 +283,46 @@ class Chain {
         /**
          * create a proposal for registrar
          */
-        let publicProps = await this.democracyPublicProps();
+        let publicProps = await self.democracyPublicProps();
         await sleep(DEFAULT_SLEEP_INTERVAL);
         if (`${publicProps.length}` === '0') {
-            // TODO_CHECK this code isn't being hit, and it's throwing build errors (remove the 2 ts-ignores to highlight the issues)
+            // TODO_CHECK self code isn't being hit, and it's throwing build errors (remove the 2 ts-ignores to highlight the issues)
             // @ts-ignore
-            await this.democracyNotePreimage(this.alice, this.identityAddRegistrar, [this.alice, registrarAccount]);
+            await self.democracyNotePreimage(self.alice, self.identityAddRegistrar, [account2registrar[`${registrarAccount.address}`]]);
             await sleep(DEFAULT_SLEEP_INTERVAL);
```

Comment:
is this because the wrong account was proposing the regisitrar?

Reply:
It's also a reason.

Reply:
If we simply call `await self.democracyNotePreimage(self.alice, self.identityAddRegistrar account2registrar[`${registrarAccount.address}`]); `, then your source code can work.